spring-cloud-generator/showcase/showcase-spring-starter/src/test/java/com/google/showcase/v1beta1/spring/EchoAutoConfigurationTests.java

```diff
@@ -189,27 +188,21 @@ void testExecutorThreadCountFromProperties() {
 
   @Test
   void testCustomTransportChannelProviderUsedWhenProvided() throws IOException {
-    when(mockTransportChannelProvider.getTransportName()).thenReturn("grpc");
-    when(mockTransportChannelProvider.getTransportChannel()).thenReturn(mockTransportChannel);
-    when(mockTransportChannel.getEmptyCallContext()).thenReturn(mockApiCallContext);
-    // Mock no-ops for ApiCallContext since this test only intends to verify override of
-    // TransportChannelProvider bean
-    when(mockApiCallContext.withCredentials(any())).thenReturn(mockApiCallContext);
-    when(mockApiCallContext.withTransportChannel(any())).thenReturn(mockApiCallContext);
-    when(mockApiCallContext.withStreamWaitTimeout(any())).thenReturn(mockApiCallContext);
-    when(mockApiCallContext.withStreamIdleTimeout(any())).thenReturn(mockApiCallContext);
-    when(mockApiCallContext.withEndpointContext(any())).thenReturn(mockApiCallContext);
-
+    InstantiatingGrpcChannelProvider channelProvider =
+        InstantiatingGrpcChannelProvider.newBuilder()
+            .setEndpoint("localhost:7469")
+            .setMaxInboundMessageSize(Integer.MAX_VALUE)
+            .build();
     contextRunner
```

Comment:
To @lqiu96 's concerns about resource closing from offline discussion:
Yes, `createChannel()` is actually called in this test. But Spring's ApplicationContextRunner (`contextRunner` in this test) is actually responsible of managing and closing all beans and data sources it manages.

Reply:
Ok, gotcha. Thanks for checking!